src/gather/api.py

```diff
@@ -72,13 +90,18 @@ def callback(scanner, inner_name, objct):
                 effective_name = inner_name
             else:
                 effective_name = name
-            scanner.registry[effective_name] = objct
+            try:
+                scanner.registry[effective_name] = scanner.strategy(scanner.registry,
+                                                                    effective_name,
+                                                                    objct)
+            except ValueError as exc:
+                scanner.please_raise = exc
```

Comment:
I wonder if this and the `tag` attribute together represent enough mass to warrant a context-like object with defined attributes.  Thoughts?